cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/scorecompiler/ExecutionPlanBuilder.java

```diff
@@ -87,24 +94,61 @@ public ExecutionPlan createFlowExecutionPlan(Flow compiledFlow) {
             throw new RuntimeException("Flow: " + compiledFlow.getName() + " has no tasks");
         }
 
-        List<ExecutionStep> taskExecutionSteps = buildTaskExecutionSteps(tasks.getFirst(), taskReferences, tasks);
+        List<ExecutionStep> taskExecutionSteps = buildTaskExecutionSteps(tasks.getFirst(), taskReferences, tasks, compiledFlow);
         executionPlan.addSteps(taskExecutionSteps);
 
         return executionPlan;
     }
 
+    public Map<String, ExecutionPlan> createBranchExecutionPlans(Flow compiledFlow) {
```

Comment:
why do we need different execution plan for each branch?
it is the same execution plan.. ins't it?